From 6e057594b0ea9547701eaac346bed71cdfb5cb94 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sat, 6 May 2017 00:58:11 -0700
Subject: [PATCH] Eleven: Fix saving/restoring user settings

*Fix saved repeat-mode and shuffle-mode settings not being restored
*Fix shake-to-play and show-album-artwork settings not being saved

Change-Id: I7ebf1d3e17c00236eb2dd6cf504a76f1dba4ee19
---
 src/com/cyanogenmod/eleven/MusicPlaybackService.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/com/cyanogenmod/eleven/MusicPlaybackService.java b/src/com/cyanogenmod/eleven/MusicPlaybackService.java
index 4fbbc63..a5b25ca 100644
--- a/src/com/cyanogenmod/eleven/MusicPlaybackService.java
+++ b/src/com/cyanogenmod/eleven/MusicPlaybackService.java
@@ -665,6 +665,9 @@ public void onCreate() {
                 PreferenceUtils.SHOW_ALBUM_ART_ON_LOCKSCREEN, true);
         setShakeToPlayEnabled(mPreferences.getBoolean(PreferenceUtils.SHAKE_TO_PLAY, false));
 
+        mRepeatMode = mPreferences.getInt("repeatmode", REPEAT_NONE);
+        mShuffleMode = mPreferences.getInt("shufflemode", SHUFFLE_NONE);
+
         registerExternalStorageListener();
 
         // Initialize the media player
@@ -2866,6 +2869,9 @@ public void setShakeToPlayEnabled(boolean enabled) {
         if (D) {
             Log.d(TAG, "ShakeToPlay status: " + enabled);
         }
+        final SharedPreferences.Editor editor = mPreferences.edit();
+        editor.putBoolean(PreferenceUtils.SHAKE_TO_PLAY, enabled);
+        editor.apply();
         if (enabled) {
             if (mShakeDetector == null) {
                 mShakeDetector = new ShakeDetector(mShakeDetectorListener);
@@ -2885,6 +2891,9 @@ public void setShakeToPlayEnabled(boolean enabled) {
      */
     public void setLockscreenAlbumArt(boolean enabled) {
         mShowAlbumArtOnLockscreen = enabled;
+        final SharedPreferences.Editor editor = mPreferences.edit();
+        editor.putBoolean(PreferenceUtils.SHOW_ALBUM_ART_ON_LOCKSCREEN, enabled);
+        editor.apply();
         notifyChange(META_CHANGED);
     }
 
